improvecentrality.py

```diff
@@ -28,278 +28,25 @@
-                )
-
-        # Filters that require more computation or are invalid for tiny
-        # nodes are excluded from cond in order to get safety and a
-        # slight performance boost from short circuiting
-
-        return (all(cond)
-                and validatenodechancapacities(n['capacities'], minchanstiers)
-                and calcavgchanage(nkey, graph, currblockheight) >= minavgchanageblks
-                and determinereliability(nkey, graph) >= minreliability
-                )
-
-    return [n['pub_key'] for n in
-            filter(lambda n: filtercandidatenodes(n, graph, filters),
-                   graph.nodes.values())
-           ]
-
-def filterrelevantnodes(graph, graphfilters):
-    minrelevantchan = graphfilters.getint('minrelevantchan')
-
-    t = time.time()
-    def filter_node(nk):
-        n = graph.nodes[nk]
-
-        cond = (
-                # A node that hasn't updated in this time might be dead
-                t - n['last_update'] < 4 * 30 *24*60*60 ,
-                graph.degree(nk) > 0, # doesn't seem to fix the issue
-            )
-
-        return all(cond)
-
-    def filter_edge(n1, n2):
-        e = graph.edges[n1, n2]
-
-        isours = mynodekey in [n1, n2]
-
-        cond = (
-                # A channel that hasn't updated in this time might be dead
-                t - e['last_update'] <=  1.5 *24*60*60,
-
-                # Remove economically irrelevant channels
-                e['capacity'] >= minrelevantchan or isours,
-                )
-
-        return all(cond)
-
-    gfilt = nx.subgraph_view(graph, filter_node=filter_node, filter_edge=filter_edge)
-
-    mostrecentupdate = 0
-    for e in graph.edges.values():
-         if e['last_update'] > mostrecentupdate:
-            mostrecentupdate = e['last_update']
-
-    print('Latest update in graph:', time.ctime(mostrecentupdate))
-    if t - mostrecentupdate > 6*60*60:
-        raise RuntimeError('Graph is more than 6 hours old, results will be innaccurate')
-
-    return gfilt
-
-## Configuration ends here
-def preparegraph(mynodekey, graphfilters):
-
-    gfull = loadgraph.lnGraph.autoload(expirehours=False)
-    for newpeer in addchannels:
-        gfull.add_edge(mynodekey, newpeer, capacity=newchannelsize, last_update=time.time())
-    for newpeer in removechannels:
-        gfull.remove_edge(mynodekey, newpeer)
-    nx.freeze(gfull)
-
-    print('Loaded graph. Number of nodes:', gfull.number_of_nodes(), 'Edges:', gfull.number_of_edges())
-    g = filterrelevantnodes(gfull, graphfilters)
-
-    print('Simplified graph. Number of nodes:', g.number_of_nodes(), 'Edges:', g.number_of_edges())
-    if g.number_of_edges() == 0:
-        raise RuntimeError('No recently updated channels were found, is describgraph.json recent?')
-
-    return g
+from GraphFilter import GraphFilter
+from CandidateFilter import CandidateFilter
+from lnGraph import lnGraph
+from fastcentrality import *
+from common_utils import *
```

Comment:
I'm not a fan of `import *` but I think it can be justified here if the utils file is exclusively used by this script.
The fastcentrality module will likely be obsoleted by my next major update, so it's ok too for now.

Reply:
Sure, I'll keep this in mind for future PR's.